ResearchAssistant/App/app.py

```diff
@@ -287,8 +287,21 @@ def stream_with_data(body, headers, endpoint, history_metadata={}):
                                 continue
 
                         if 'error' in lineJson:
-                            yield format_as_ndjson(lineJson)
-                        
+                            error_code_value = lineJson.get('error', {}).get('code', '')
+                            error_message = format_as_ndjson(lineJson)
+                            inner_error_code_value = extract_value('code', error_message)
+                            inner_error_status_value = extract_value('status', error_message)
+                            if inner_error_code_value == 'content_filter' and inner_error_status_value == '400':
```

Comment:
@Pavan-Microsoft @Roopan-Microsoft @Prajwal-Microsoft  I have a question so if we ask unethical question does the API response comes back as 400 from OpenAI. Asking this for my general knowledge sake?

Reply:
> @Pavan-Microsoft @Roopan-Microsoft @Prajwal-Microsoft I have a question so if we ask unethical question does the API response comes back as 400 from OpenAI. Asking this for my general knowledge sake?

Yes, we do receive a 400 error. Here's a sample exception:

{'error': {'requestid': '8bdecd03-7617-496e-9835-aaaaaaaa', 'code': 400, 'message': 'An error occurred when calling Azure OpenAI: Server responded with status 400. Error message: {\'error\': {\'message\': "The response was filtered due to the prompt triggering Azure OpenAI\'s content management policy. Please modify your prompt and retry. To learn more about our content filtering policies please read our documentation: https://go.microsoft.com/fwlink/?linkid=2198766", \'type\': None, \'param\': \'prompt\', \'code\': \'content_filter\', \'status\': 400, \'innererror\': {\'code\': \'ResponsibleAIPolicyViolation\', \'content_filter_result\': {\'hate\': {\'filtered\': True, \'severity\': \'medium\'}, \'self_harm\': {\'filtered\': False, \'severity\': \'safe\'}, \'sexual\': {\'filtered\': False, \'severity\': \'safe\'}, \'violence\': {\'filtered\': False, \'severity\': \'low\'}}}}}'}}